homeassistant/components/fritz/sensor.py

```diff
@@ -39,37 +39,37 @@ def _retrieve_uptime_state(status: FritzStatus, last_value: str) -> str:
 
 def _retrieve_external_ip_state(status: FritzStatus, last_value: str) -> str:
     """Return external ip from device."""
-    return status.external_ip
+    return status.external_ip  # type: ignore[no-any-return]
 
 
-def _retrieve_kib_s_sent_state(status: FritzStatus, last_value: str) -> str:
+def _retrieve_kib_s_sent_state(status: FritzStatus, last_value: str) -> float:
     """Return upload transmission rate."""
-    return round(status.transmission_rate[0] * 8 / 1024, 1)
+    return float(round(status.transmission_rate[0] * 8 / 1024, 1))
```

Comment:
Isn't `round` return float already?

Reply:
Nope, please check docs: [round()](https://docs.python.org/3/library/functions.html?highlight=round#round)

"The return value is an integer if ndigits is omitted or None. Otherwise the return value has the same type as number."

Simone

Reply:
Exactly. The input type is `float` because of division and you have `ndigits` specified.

Reply:
```
def round(number: SupportsRound[_T], ndigits: int) -> _T: ...
```
Indeed, should be returning a float already in its typing.

Reply:
Unfortunately `status.transmission_rate[0]` is Any and mypy complains about such situation. This is the only reason why I added the float.

Reply:
Would this be a better solution in that case?

```suggestion
    transmission_rate: float = status.transmission_rate[0]
    return round(transmission_rate * 8 / 1024, 1)
```

Reply:
I think it's better to just add `# type: ignore` until `FritzStatus` will become typed.

Reply:
So which is the implementation you prefer ?

Reply:
Agreed with @KapJI, with the small addition to make it specific.
E.g., `# type: ignore[no-any-return]`

Reply:
Done!